journal.py

```diff
@@ -16,78 +16,83 @@
+    created = sa.Column(
+        sa.DateTime, nullable=False, default=datetime.datetime.utcnow
+    )
 
-UPDATE_ENTRY = """UPDATE entries SET title=%s, text=%s WHERE id=%s
-"""
+    def __repr__(self):
+        return u"{}: {}".format(self.__class__.__name__, self.title)
 
-logging.basicConfig()
-log = logging.getLogger(__file__)
+    @classmethod
+    def all(cls):
+        return DBSession.query(cls).order_by(cls.created.desc()).all()
 
+    @classmethod
+    def newest_entry(cls):
+        return DBSession.query(cls).order_by(cls.created.desc()).first()
 
-def connect_db(settings):
-    """Return a connection to the configured database"""
-    return psycopg2.connect(settings['db'])
+    @classmethod
+    def by_id(cls, id):
+        return DBSession.query(cls).filter(cls.id == id).one()
 
+    @classmethod
+    def from_request(cls, request):
+        title = request.params.get('title', None)
+        text = request.params.get('text', None)
+        created = datetime.datetime.utcnow()
+        new_entry = cls(title=title, text=text, created=created)
+        DBSession.add(new_entry)
 
-def init_db():
-    """Create database dables defined by DB_SCHEMA
+    def update_from_request(self, request):
+        self.title = request.params.get('title', None)
+        self.text = request.params.get('text', None)
 
-    Warning: This function will not update existing table definitions
-    """
-    settings = {}
-    settings['db'] = os.environ.get(
-        'DATABASE_URL', 'dbname=learning_journal user=chatzis'
-    )
-    with closing(connect_db(settings)) as db:
-        db.cursor().execute(DB_SCHEMA)
-        db.commit()
+    def render_markdown(self):
+        return markdown.markdown(
+            self.text, extensions=['codehilite', 'fenced_code'])
 
+    def convert_strftime(self):
+        return self.created.strftime('%b %d, %Y')
 
-@subscriber(NewRequest)
-def open_connection(event):
-    request = event.request
-    settings = request.registry.settings
-    request.db = connect_db(settings)
-    request.add_finished_callback(close_connection)
+    def json(self):
+        return {'title': self.title,
+                'text': self.render_markdown(),
+                'created': self.created.strftime('%b %d, %Y'),
+                'id': self.id}
 
+    def json_edit_get(self):
+        return {'title': self.title,
+                'text': self.text,
+                'created': self.created.strftime('%b %d, %Y'),
+                'id': self.id}
 
-def close_connection(request):
-    """close the database connection for this request
 
-    If there has been an error in the processing of the request, abort any
-    open transactions.
-    """
-    db = getattr(request, 'db', None)
-    if db is not None:
-        if request.exception is not None:
-            db.rollback()
-        else:
-            db.commit()
-        request.db.close()
+logging.basicConfig()
+log = logging.getLogger(__file__)
```

Comment:
I think this pair of lines is now in the file twice.  Pick one.  Probably the first, not this one.